bentoml/marshal/marshal.py

```diff
@@ -352,22 +373,41 @@ async def _batch_handler_template(self, requests, api_route):
                 for i in merged
             )
 
-    def async_start(self, port):
-        """
-        Start an micro batch server at the specific port on the instance or parameter.
-        """
-        marshal_proc = multiprocessing.Process(
-            target=self.fork_start_app, kwargs=dict(port=port), daemon=True,
-        )
-        marshal_proc.start()
-        logger.info("Running micro batch service on :%d", port)
-
-    def make_app(self):
+    def make_app(self) -> "Application":
         from aiohttp.web import Application
+        from aiohttp import hdrs
+
+        methods = hdrs.METH_ALL.copy()
+
+        if self.enable_access_control:
+            # ref: https://github.com/aio-libs/aiohttp-cors/issues/241
+            methods.remove(hdrs.METH_OPTIONS)
 
         app = Application(client_max_size=self.max_request_size)
-        app.router.add_view("/", self.relay_handler)
-        app.router.add_view("/{path:.*}", self.request_dispatcher)
+
+        for method in methods:
+            app.router.add_route(method, "/", self.relay_handler)
+            app.router.add_route(method, "/{path:.*}", self.request_dispatcher)
+
+        if self.enable_access_control:
+            assert (
+                self.access_control_allow_origin is not None
+            ), "To enable cors, access_control_allow_origin must be set"
+            assert self.access_control_options is not None
```

Comment:
add a similar assert error message this line?

Reply:
The `self.access_control_options` won't be None when enabled is True. This line is to make type checker happy.